src/components/AvatarWithImagePicker.js

```diff
@@ -232,158 +224,168 @@ class AvatarWithImagePicker extends React.Component {
-                                accessibilityRole={CONST.ACCESSIBILITY_ROLE.IMAGEBUTTON}
-                                accessibilityLabel={this.props.translate('avatarWithImagePicker.editImage')}
-                                disabled={this.state.isAvatarCropModalOpen}
-                                ref={this.anchorRef}
-                            >
-                                <View>
-                                    {this.props.source ? (
-                                        <Avatar
-                                            containerStyles={styles.avatarLarge}
-                                            imageStyles={[styles.avatarLarge, styles.alignSelfCenter]}
-                                            source={this.props.source}
-                                            fallbackIcon={this.props.fallbackIcon}
-                                            size={this.props.size}
-                                            type={this.props.type}
-                                        />
-                                    ) : (
-                                        <DefaultAvatar />
-                                    )}
-                                </View>
-                                <View style={[styles.smallEditIcon, styles.smallAvatarEditIcon]}>
-                                    <Icon
-                                        src={Expensicons.Camera}
-                                        width={variables.iconSizeSmall}
-                                        height={variables.iconSizeSmall}
-                                        fill={themeColors.textLight}
+    };
+
+    const hideAvatarCropModal = () => {
+        setIsAvatarCropModalOpen(false);
+    };
+
+    /**
+     * Create menu items list for avatar menu
+     *
+     * @param {Function} openPicker
+     * @returns {Array}
+     */
+    const createMenuItems = (openPicker) => {
+        const menuItems = [
+            {
+                icon: Expensicons.Upload,
+                text: translate('avatarWithImagePicker.uploadPhoto'),
+                onSelected: () => {
+                    if (Browser.isSafari()) {
+                        return;
+                    }
+                    openPicker({
+                        onPicked: showAvatarCropModal,
+                    });
+                },
+            },
+        ];
+
+        // If current avatar isn't a default avatar, allow Remove Photo option
+        if (!isUsingDefaultAvatar) {
+            menuItems.push({
+                icon: Expensicons.Trashcan,
+                text: translate('avatarWithImagePicker.removePhoto'),
+                onSelected: () => {
+                    setError(null, {});
+                    onImageRemoved();
+                },
+            });
+        }
+        return menuItems;
+    };
+
+    return (
+        <View style={StyleSheet.flatten([styles.alignItemsCenter, style])}>
```

Comment:
Please re-test this on all of the platforms after making the final changes, currently this is crashing (`StyleSheet` isn't imported)

https://github.com/Expensify/App/assets/9059945/fe709ecc-fc8a-4f50-9439-3ccf21706e08

Reply:
@eVoloshchak please find the videos below

macOS desktop
https://github.com/Expensify/App/assets/18525360/90b71053-79ca-4ad8-a22b-d3441bcbf5fe

macOS safari
https://github.com/Expensify/App/assets/18525360/76af6956-a06f-4358-bc04-e1af0b22dfd0

macOS chrome
https://github.com/Expensify/App/assets/18525360/c705c44e-3769-4663-847c-5f7719ec4890

ios
https://github.com/Expensify/App/assets/18525360/80af5907-d25f-430d-b0e1-a6cb0bb68bb4

ios safari
https://github.com/Expensify/App/assets/18525360/c78d8de8-6bb0-436e-96b6-ec803a32bd8f

android
https://github.com/Expensify/App/assets/18525360/9284b056-92fb-4299-9496-713efa8ef43b

android chrome
https://github.com/Expensify/App/assets/18525360/38c8efc3-9683-45cb-acae-18e8698f7d56